src/inputs/internal/SelectFieldBase.tsx

```diff
@@ -15,8 +15,10 @@ import { areArraysEqual } from "src/utils";
 export interface BeamSelectFieldBaseProps<O, V extends Value> extends BeamFocusableProps, PresentationFieldProps {
   /** Renders `opt` in the dropdown menu, defaults to the `getOptionLabel` prop. `isUnsetOpt` is only defined for single SelectField */
   getOptionMenuLabel?: (opt: O, isUnsetOpt?: boolean) => string | ReactNode;
-  getOptionValue: (opt: O) => V;
-  getOptionLabel: (opt: O) => string;
+  /** Defaults to `id ?? value` */
+  getOptionValue?: (opt: O) => V;
+  /** Defaults to `displayName || label || name || value` */
+  getOptionLabel?: (opt: O) => string;
```

Comment:
Thought: Some of this should already be handled by the `SelectField` overload:

```
export function SelectField<O, V extends Value>(props: SelectFieldProps<O, V>): JSX.Element;
export function SelectField<O extends HasIdAndName<V>, V extends Value>(
  props: Optional<SelectFieldProps<O, V>, "getOptionValue" | "getOptionLabel">,
): JSX.Element;
export function SelectField<O, V extends Value>(
  props: Optional<SelectFieldProps<O, V>, "getOptionLabel" | "getOptionValue">,
```

I.e. it conditionally `Optionalize`s `getOptionValue` and `getOptionLabel` only when `O` has an id and name.

Granted, that was very inflexible, but it is more type-safe; can you try working the `AtLeastOne` trick as a replacement for `O extends HasIdAndName`, so that ideally we get both the increased flexibility of "use any one of these N fields" while not having to fallback on runtime checks for callers not passing the properties when they should?

(I had tried making this more flexible in the past, but didn't have the `AtLeastOne` trick, so it led to ~an egregious amount of overloads, like 1 overload per guessed label, 1 overload per guessed value, and so just gave up.)